### PR TITLE
chart: enable cert-manager

### PR DESCRIPTION
**Pull request by alejandroEsc**
Signed-off-by: alejandroEsc <jaescobar.cell@gmail.com>

**What this PR does / why we need it**:
Enables cert-manager integration to handle cert creation for the webhook

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by makkes** [comment]
In accordance with what we committed ourselves to in https://github.com/kubernetes-sigs/kubefed/pull/1270#issuecomment-671789700 let's get sign-off from a different organization's member.

@alejandroEsc, would you mind assigning this to one of the other members in https://github.com/kubernetes-sigs/kubefed/blob/79cb7e20d3d388ee8a5c12e3ed7cdd51a8b14b99/OWNERS#L10, please?

**Comment by alejandroEsc** [comment]
> In accordance with what we committed ourselves to in [#1270 (comment)](https://github.com/kubernetes-sigs/kubefed/pull/1270#issuecomment-671789700) let's get sign-off from a different organization's member.
> 
> @alejandroEsc, would you mind assigning this to one of the other members in
> 
> https://github.com/kubernetes-sigs/kubefed/blob/79cb7e20d3d388ee8a5c12e3ed7cdd51a8b14b99/OWNERS#L10
> 
> , please?
> /cc @irfanurrehman fyi

I was merely replying to what the robots instructions.

File: charts/kubefed/Chart.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
 apiVersion: v2
 description: KubeFed helm chart
 name: kubefed
-version: 0.0.3
+version: 0.0.4
```
Comment (irfanurrehman) [inline charts/kubefed/Chart.yaml:4]: Is this change in anticipation of a release? Wouldn't this match the latest released version and better be changed as part of the release process, or a PR preparing for a release?

File: charts/kubefed/charts/controllermanager/templates/webhook.yaml [hunk 2]
```diff
@@ -137,3 +150,49 @@ type: kubernetes.io/tls
...
+metadata:
+  name: {{ .Release.Name }}-certificate
+spec:
+  secretName: kubefed-admission-webhook-serving-cert
+  duration: 8760h # 1y
+  issuerRef:
+    name: {{ .Release.Name }}-issuer
+  dnsNames:
+    - kubefed-admission-webhook
+    - {{ $altName1 }}
```
Comment (irfanurrehman) [inline charts/kubefed/charts/controllermanager/templates/webhook.yaml:196]: Where are these supposed to be filled from ?

File: charts/kubefed/values.yaml [hunk 3]
```diff
@@ -27,6 +27,7 @@ controllermanager:
     FederatedIngress:
   
   controller:
+    annotations: {}
```
Comment (irfanurrehman) [inline charts/kubefed/values.yaml:30]: Reason for this? Does omitting this break something?

**Comment by irfanurrehman** [comment]
Thanks @alejandroEsc for this piece of work. A few nits/questions, else it a good to go from me.

> > In accordance with what we committed ourselves to in [#1270 (comment)](https://github.com/kubernetes-sigs/kubefed/pull/1270#issuecomment-671789700) let's get sign-off from a different organization's member.
> > @alejandroEsc, would you mind assigning this to one of the other members in
> > https://github.com/kubernetes-sigs/kubefed/blob/79cb7e20d3d388ee8a5c12e3ed7cdd51a8b14b99/OWNERS#L10
> > 
> > , please?
> > /cc @irfanurrehman fyi
> 
> I was merely replying to what the robots instructions.

@makkes IMO its not necessary to assign the PR to a person from different org, and anybody can review the PR. I did not mean that only a person from another org can review the PR (Hope that is not what is understood.. :) ).  My suggestion was to ensure that the final merge is done by somebody who carries a possible different set of opinion.

File: charts/kubefed/charts/controllermanager/templates/webhook.yaml (hunk 2 above)
Comment (alejandroEsc) [reply charts/kubefed/charts/controllermanager/templates/webhook.yaml:196]: from the top of the file: `{{- $altName1 := printf "kubefed-admission-webhook.%s" .Release.Namespace }}`

File: charts/kubefed/Chart.yaml (hunk 1 above)
Comment (alejandroEsc) [reply charts/kubefed/Chart.yaml:4]: force of habit, apologies.

File: charts/kubefed/values.yaml (hunk 3 above)
Comment (alejandroEsc) [reply charts/kubefed/values.yaml:30]: would be nice to inject annotations to the deployments.

**Comment by alejandroEsc** [comment]
@irfanurrehman thank you so much for reviewing!

**Comment by irfanurrehman** [comment]
Thanks for doing this @alejandroEsc. `lgtm`
Leaving it to another set of eyes to review/merge. (@makkes it could be you, if you are happy with this PR too, I will merge).

**Pull request merged**